reminders.py

```diff
@@ -0,0 +1,350 @@
+    except:
+        exit(1);
+
+    # Exit with failure if bad status code returned
+    if (response.status_code != requests.codes.ok):
+        exit(1);
+
+    return;
+
+
+# Stutus Bitbucket
+def statusBitbucket():
+    url = BITBUCKET_URL + STATUS_ENDPOINT;
+    queryBitbucket(url);
+
+
+# Fetch Pull Requests (defaults to OPEN PRs only)
+def fetchOpenPRs():
+    idVals = [];
+    url = BITBUCKET_URL + REST_ENDPOINT + PULL_REQUESTS;
+    response = queryBitbucket(url);
+
+    # Iterate through response to get PR IDs
+    content = response.json();
+    for key, prArray in content.items():
+        if (key == 'values' and prArray):
+            for pr in prArray:
+                idVals.append(pr['id']);
+            break;
+
+    # No Open PRs, exit
+    if not idVals:
+        exit(0)
+
+    return idVals;
+
+
+# Using datetime objects inspect PR age and update duration
+def analyzePRTimes(prID, pullRequestJSON):
+    createdDate = '';
+    updatedDate = '';
+    runChecks = False;
+
+    # Store creation and updated dates
+    for key, value in pullRequestJSON.items():
+        if (key == 'createdDate'):
+            createdDate = datetime.datetime.fromtimestamp(value);
+        elif (key == 'updatedDate'):
+            updatedDate = datetime.datetime.fromtimestamp(value);
+
+    # PR should not be missing these values, exit with failure
+    if not createdDate or not updatedDate:
+        exit(1)
+
+    # Store time differences and period last updated
+    prAge = TODAY - createdDate;
+    updatePeriod = updatedDate - createdDate;
+
+    # Check if PR is older than 10 days
+    if (prAge > TENDAYS):
+        suggestDecline(prID);
+    elif (updatePeriod > TWODAYS):
+        runChecks = True;
+
+    return runChecks;
+
+
+# Using datetime objects inspect PR activity age
+def analyzeActionTimes(prID, url):
+    runChecks = True;
+    activityUrl = url + ACTIVITIES;
+    response = queryBitbucket(activityUrl);
+
+    # Iterate through response to get activity content
+    content = response.json();
+    for key, activityArray in content.items():
+        if (key == 'values' and activityArray):
+            for activity in activityArray:
+                action = activity['action'];
+                createdDate = datetime.datetime.fromtimestamp(activity['createdDate']);
+
+                # Check if desired activity occured < 2 days ago
+                if (action == 'COMMENTED' or action == 'REVIEWED' or action == 'APPROVED'):
+                    activityAge = TODAY - createdDate;
+                    if (activityAge < TWODAYS):
+                        runChecks = False;
+                        break;
+            break;
+
+    return runChecks;
+
+
+# Alert author to decline PR
+def suggestDecline(prID):
+    delim = "\n";
+    msg = [];
+
+    # Prepare comment
+    msg.append("Good day sir,");
+    msg.append("It appears your review has gone rather stale!");
+    msg.append("May I suggest declining and performing further work?");
```

Comment:
Can this script make me dinner or at least a coffee?

Reply:
orders ubereats on weekends for us with a I'm feeling lucky option 😛